tests/ui/tests/lib/capabilities.js

```diff
@@ -0,0 +1,29 @@
+define(['intern/dojo/node!leadfoot/keys'], function(keys) {
+
+    return {
+
+        getBrowserName: function(remote) {
+            // Returns the browser of the current session
+
+            return remote.session.capabilities.browserName.toLowerCase();
+        },
+
+        crossbrowserConfirm: function(remote) {
+            // Firefox and Chrome react more reliably to elements with the ENTER key pressed
+            // Safari, on the other hand, only responds to click
+
+            var gbn = this.getBrowserName(remote);
+
+            return function(element) {
+                if(gbn === 'safari') {
```

Comment:
Sucks but not as bad as you think -- we control the names of the browsers so it's fine.  We're writing for Selenium/node, not the browser.